### ISSUE TITLE
Connexion timeout

### ISSUE THREAD

**Issue by cedricalpatch** [ImagicTheCat/vRP#88]
Hello ,

I have lot of problem for lot of players to connect to the server... This is RPF Studio ;)
We have timeout after 15sc for lot of players if you have allready have this problem?

Cedric

**Comment by cedricalpatch** [comment]
Other, I need change the MySql.Data.dll by essentialmode MySql.Data.dll and all work, or not I cannot connect to sql

**Comment by ImagicTheCat** [comment]
I heard something like this about another server, but I can't see why vRP will cause this, can you confirm this is only related to vRP by removing it and see if the problem is gone ?
I don't understand your question about MySql.Data.dll, vRP use recent features of the mysql connector, it's the version 4.5, you can't use both the mysql of essentialmode and the mysql of vRP on the same server.

**Comment by Geronimin** [comment]
Yes @ImagicTheCat Problemconfirmed as we released the server yesterday PM and we hadn't the opportunity to be more than 15/24 (more or less), and we where on essentialmode with full + waiting queue.
Symptoms are: It seems like some players are in conflict together (1 player and another one reciprocly lets say A and B), and when A is online, B is always getting in a 'different session' as soon he finished to connect (precisly when he get his clothes and rights syncronised)(he is alone but online in the serv's console and he's able to communicate by the tchat). The strange thing is that it's only when we get over 12-13p connected.

I'm here if you want ppl to test things or ask any question.

Kind regards, Muto.

PS: Reboot players entries in bdd; done. 
     Reinstall fivem for the players getting issues: done.
     We saw a strange ip in db something like 20:34:15.
     It seems like it's defined persons (i means not random, always the sames).

**Comment by ImagicTheCat** [comment]
Thanks for the informations, when you say you were on essentialmode, you mean also vRP or it's an unrelated issue ?

**Comment by cedricalpatch** [comment]
Hello,

I have try with one other server an not problem... I think this come from the proxy/tunel or I dont know??

**Comment by cedricalpatch** [comment]
And the data base is alown just for vrp, not essential or any othe script... :(
After last update from Fivem no problem

**Comment by cedricalpatch** [comment]
the version of Vrp is last yes ;)

**Comment by Geronimin** [comment]
@ImagicTheCat Not sure i get what you wanted to ask me, but if i well understood, but i was on essential with es_freeroam and nothing about vRP, then i started again from scratch (fivem server) and i installed vrp and dev on it (so there is no more essential & es_freeroam)

**Comment by cedricalpatch** [comment]
testing 👍 

Change ip box from client
remove all fivem and GTA

Sever side change port for 8080 , 4664 , 30120 , 30200 ...
Restart server with just base of VRP not any modif.

Still same problem :(

**Comment by Bandit33800** [comment]
hi,
The timeout is related to an internet connection too weak

**Comment by cedricalpatch** [comment]
Yes I think too is the problem...
So any solution because lot of players havent best connexion? on 25 players I have juste 10 can connect :(

**Comment by cedricalpatch** [comment]
With GTA Netwok with proxy / tunel is not problem the low connexion so maybe we can found solution for fivem=?

**Comment by ImagicTheCat** [comment]
Proxy and tunnels are just a layer on top of TriggerEvent, I don't think the bandwidth of TriggerEvent is so low that you can be disconnected if receiving too much events.

@Geronimin: so you say you reproduced the issue without vRP ? So it's a FiveM issue not a vRP issue ?

**Comment by Geronimin** [comment]
@ImagicTheCat Nonono sorry if i've been explaining it badly ^^.
Let's try again: Im' saying that the issues has shown it head when we passed on vRP, before we were on fivem+ essential + es_freeroam, with issues of instability and frequent disconects and crash but randomly.
We we started from scratch and using vRP, we've been impressed of the stability that vRP offers, but now we have the issues i was explaining @ the biggining of the topic.

Telle me again if you didn't get it, no pb if i must precise more or reformulate

**Comment by ImagicTheCat** [comment]
Okay, add any informations you have about the problem on this issue, so we can solve it. 
Things we need to check/validate:
* Timed out people are always the same, but only after a specific amount of players already connected, on any vRP server (if true, we need to know their connections performance)
* What is the console message, what is the client message, how many seconds after spawning before being timed out
* Can you give item to someone in another instance when at the same place of the map ?
* Can you send sms to people in other instances ?
* People in other instances are able to change their clothes ?

**Comment by Geronimin** [comment]
<ul>

> <li>Timed out people are always the same, but only after a specific amount of players already connected, on any vRP server (if true, we need to know their connections performance)</li>
<li>Timed out ppl aren't the same, i truely think than it's just when we're over a certain amount of player connected, the thing i tried to explain you that made you thought that is that when we're over a certain amount of player then they are always the same (new persons connecting after the fictiv limit are guetting issues, but they are not making people connected before we reach the limit getting timming out)</li>
<li>There is no console errors (server side) just connect, and disconection (when player disconect, not when he gets in an other instance. Unfortunately it's only when i have over 10-15 players connected so basically it's hard to say if i don't have messages, but im sure there is no error line. 
I didn't check by myself but 2 persons told me that there is no errors in client console</li>

</ul>

**Comment by ImagicTheCat** [comment]
Just paste any strange message in server console (net, host elected, etc).

**Comment by ImagicTheCat** [comment]
Also, do they freeze client-side before be timed out ?

**Comment by Geronimin** [comment]
![proof](https://cloud.githubusercontent.com/assets/28099686/26215469/b9bcb95a-3c00-11e7-8345-f538fd579663.png)

I got a proof, the blue line happened just when this player finished to load his datas (skins, power level), just next to it you got the servs perf on task manager.

**Comment by ImagicTheCat** [comment]
Thanks for the report. Be careful to hide the IP address of your players when posting screenshots, github is public and there are some malicious ways to use them.
So, when someone is instanced, the message "Received a vote..." appear, right ? This is not a new issue, https://forum.fivem.net/t/disconnecting-players-received-a-vote-to-change-host/12899 .

I don't know what we can learn from the blue line, but can you do a speed test of your server connection and post the result here ?

To solve this issue, I think we need first to understand what is causing this with FiveM, then try to understand why vRP increase the probabilities.
I don't think this is a MySQL blocking problem, because I saw the issue on a server with 3 people and did some benchmarks, where I couldn't have the issue myself. Something is making players to leave the server instance, but I don't know why. This is related to the fact that the number of slots doesn't exceed 32 players to avoid instance instability, so something is making it worst.

**Comment by Geronimin** [comment]
Ok, i think that we found out what's about ! There is a lot of proof, but we can't be sure 90%. Anyway i can explain you on vocal if you want because it's very long, but it's seems like there is an order due to our ids in database. 
1) there are slots reserved for player id < X (where X represent the barreer of who can connect on reserved slots [<X] and who can't [>X]). X could be 10 where 14 ~= the ledge where timouts starts 
2) For the not reserved slots available, there is a priority of connection due to a test on users id's ( let's say 24 slots, 10 reserved = 14slots. So imagine that we are 14/24 on serv, the maximum value of id is 101. If someone with id over 101 will get in an other instance as soon as he finished to load his data [ And id btw ! coincidence again :) ? ]. In an other hand, if someone under id 101 tries to connect, then he will be able to connect and player id = 101 will be instantly in a new instance as soon as player id<101 finished retrieving data)

**Comment by cedricalpatch** [comment]
we cannot change the time out by 30sc or I dont know ... I have lot of time out, maybe 8/10 players :(
Me VRP server is alown...

**Comment by cedricalpatch** [comment]
Same if I have some players ingame... Now we are 4 players and I see time out time out time out...

**Comment by MagnanQC** [comment]
Are you able to send me a ping command ?

**Comment by cedricalpatch** [comment]
ping command ? The ping of my server is arrond 45

**Comment by cedricalpatch** [comment]
@ImagicTheCat still not solution?

**Comment by ImagicTheCat** [comment]
To find a solution I need to reproduce the problem as much as possible, but I never experienced it myself.

**Comment by cedricalpatch** [comment]
https://discord.gg/KNjJr4J

**Comment by Geronimin** [comment]
@ImagicTheCat Did you thought about what i said ?

**Comment by Geronimin** [comment]
![prro1](https://cloud.githubusercontent.com/assets/28099686/26275361/fa2d7362-3d5e-11e7-8390-56aa642b0835.png)

Exmeple of id 100 and more getting disconection when id 21 come ig
Now the vote is succes

**Comment by Armagryzor** [comment]
Same problem too .. but just before time out of the player , i can see the server upload bandwidth go to 10mb/s and + !! and this just few seconds before than the player is time out..

**Comment by ImagicTheCat** [comment]
How the hell do you have a 90 Mb/s input speed ? Is this a comma in European format ? Most people have a weak upload connection, ask them what are their upload speed when timing out. Also, do the players freeze before timing out ?

**Comment by ImagicTheCat** [comment]
You all agree that the issue appear just before the teleportation succeed ? (there is a 5 seconds delay before teleporting).

**Comment by cedricalpatch** [comment]
The problem is fixed I think for my! 
Thanks a lot for your best work ;)

**Comment by ImagicTheCat** [comment]
We will see if it's the case for everyone.

**Comment by cedricalpatch** [comment]
ok no problem ;)

**Comment by SnaiCrys** [comment]
Random crashes
![fffddcc7c5844dd49a61d30aa49b744b](https://cloud.githubusercontent.com/assets/4284827/26463077/d4a3e866-418b-11e7-9f40-eb704998a440.jpeg)

**Comment by ImagicTheCat** [comment]
It's not the same issue, your server is unstable, causing timeout. Open a new issue, set debug to true and give me the output.

**Comment by cedricalpatch** [comment]
after you last update for timeout I have also set false for "# set to false to require players to be signed in to Steam" in citmp-server.yml and this finish the time out every time...

Hop this help some one ;)
